CMakeLists.txt

```
@@ -138,8 +139,17 @@ if(NOT CORENRN_ENABLE_TIMEOUT)
   add_definitions("-DDISABLE_TIMEOUT")
 endif()
 
+if ( CORENRN_ENABLE_SONATA_REPORTS AND NOT CORENRN_ENABLE_REPORTINGLIB )
+  message(FATAL_ERROR "Can't have SONATA reports without enabling reportinglib")
+endif()
+
 if(CORENRN_ENABLE_REPORTINGLIB)
-  find_package(reportinglib REQUIRED)
+  if(CORENRN_ENABLE_SONATA_REPORTS)
+    find_package(reportinglibsonata REQUIRED)
+    add_definitions("-DENABLE_SONATA_REPORTS")
```

```suggestion
    add_definitions("-DENABLE_SONATA_REPORTS -DENABLE_REPORTING")
```
something like this I meant!